### PR TITLE
MRG: Add ability to load NIRX data by passing path to header file

### PR DESCRIPTION
**Pull request by rob-luke**
#### Reference issue
https://github.com/cbrnr/mnelab/issues/144#issuecomment-623066162

#### What does this implement/fix?
Allows user to pass the path to a NIRX header file to load data, rather than passing the path to the directory.

This was suggested as a way to easily add data loading to MNELAB (see comment above).

### CODE REVIEW

File: mne/io/nirx/nirx.py [hunk 1]
```diff
@@ -36,6 +37,10 @@ def read_raw_nirx(fname, preload=False, verbose=None):
     --------
     mne.io.Raw : Documentation of attribute and methods.
     """
+    if fname.endswith('.hdr'):
+        return read_raw_nirx(op.dirname(op.abspath(fname)),
+                             preload=preload, verbose=verbose)
+
     return RawNIRX(fname, preload, verbose)
```
Comment (agramfort) [inline mne/io/nirx/nirx.py:44]: I woudl do this RawNIRX init and do:

```
if fname.endswith('.hdr'):
    fname  = op.dirname(op.abspath(fname))
```

Comment (rob-luke) [reply mne/io/nirx/nirx.py]: Thanks

**Comment by cbrnr** [comment]
Can you also add a what's new entry please?

**Comment by rob-luke** [comment]
@larsoner could you please review. I think the CI failure is unrelated.

**Comment by agramfort** [comment]
thx @rob-luke

**Pull request merged**